laplace/lllaplace.py

```diff
@@ -103,12 +103,13 @@ def fit(self, train_loader, override=True):
         self.model.eval()
 
         if self.model.last_layer is None:
-            X, _ = next(iter(train_loader))
+            # Save an example batch for when loading the serialized Laplace
+            self.X, _ = next(iter(train_loader))
```

Comment:
Could it make sense to save this rather as a shape and type instead of an actual sample that could be quite big? If it's too much effort, keeping it as is might be fine.

Reply:
Good catch. I think it's better to do this since `X` is not alway a tensor. E.g. for HuggingFace models, this is `UserDict` with certain keys. For `likelihood='reward_modeling'`, this is also a `UserDict` but with other set of keys.

Reply:
Ok so in this case one would have to apply something like a map over the userdict so it seems tricky. Seems fine to go this way then. I was rather thinking of other image or video problems where the input is on the order of the parameters. llms have very small inputs indeed.

Reply:
Yeah, besides, `self.X` is on `cpu`. So, at least it won't incur burden on GPU. We can revisit this if specific use cases are requested by users.